jakarta-ee-messaging/src/main/kotlin/io/qalipsis/examples/jakarta/JakartaProduceAndConsumeFromTopic.kt

```diff
@@ -0,0 +1,129 @@
+    fun scenarioProduceAndConsumeFromTopic() {
+
+        //we define the scenario, set the name, number of minions and rampUp
+        scenario {
+            minionsCount = ScenarioConfiguration.NUMBER_MINION
+            profile {
+                regular(periodMs = 1000, minionsCountProLaunch = minionsCount)
+            }
+        }
+            .start()
+            .jackson() //we start the jackson step to fetch data from the csv file. we will use the csvToObject method to map csv entries to list of utils.BatteryState object
+            .csvToObject(BatteryState::class) {
+
+                classpath("battery-levels.csv")
+                // we define the header of the csv file
+                header {
+                    column("deviceId")
+                    column("timestamp")
+                    column("batteryLevel").integer()
+                }
+                unicast()
+            }
+            .map { it.value } // we transform the output of the CSV reader entries to utils.BatteryState
+            .jakarta()
+            .produce {
+
+                connect {
+                    ActiveMQConnectionFactory(
+                        ServerConfiguration.SERVER_URL,
+                        ServerConfiguration.CONTAINER_USERNAME,
+                        ServerConfiguration.CONTAINER_PASSWORD
+                    ).createTopicConnection()
+                }
+
+                session {
+                    it.createSession(false, Session.AUTO_ACKNOWLEDGE)
+                }
+
+                records { _, input ->
+                    listOf(
+                        JakartaProducerRecord(
+                            destination = Topic(ServerConfiguration.TOPIC_NAME),
+                            messageType = JakartaMessageType.BYTES,
+                            value = objectMapper.writeValueAsBytes(input)
+                        )
+                    )
+                }
+            }
+            .map {
+                it.input
+            }
+            .innerJoin(
+                using = { correlationRecord ->
+                    println(" Using ${correlationRecord.value} ")
+                    correlationRecord.value.primaryKey()
+                },
+
+                on = {
+                    it.jakarta().consume {
+                        topics(ServerConfiguration.TOPIC_NAME)
+                        topicConnection {
+                            ActiveMQConnectionFactory(
+                                ServerConfiguration.SERVER_URL,
+                                ServerConfiguration.CONTAINER_USERNAME,
+                                ServerConfiguration.CONTAINER_PASSWORD
+                            ).createTopicConnection()
+                        }
+                    }
+                        .deserialize(JakartaJsonDeserializer(targetClass = BatteryState::class))
+                        .map { result ->
+                            println("on ${result.record.value}")
```

Comment:
I add missed it. I think that the println were added for debugging. Please remove them before you squash and merge.